### ISSUE TITLE
Cluster resources inadvertently deleted during upgrade 2.7.3 -> 2.8.3

### ISSUE THREAD

**Issue by jstolfer** [argoproj/argo-cd#16363]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

We manage multiple eks clusters with each cluster being an individual application pointed to respective kustomization.yaml to deploy tooling to each cluster.

During the upgrade process a single cluster had all resources deleted. It appeared as though argocd thought the kustomize file it was pointed was in an un-buildable state.

No changes were made to the cluster kustomize repo during this time.

**Expected behavior**

Upgrade to complete with no changes made to managed applications/clusters

**Version**

argocd: v2.8.4+c279299.dirty
  BuildDate: 2023-09-13T20:25:25Z
  GitCommit: c27929928104dc37b937764baf65f38b78930e59
  GitTreeState: dirty
  GoVersion: go1.21.1
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.8.3+77556d9
  BuildDate: 2023-09-07T16:05:43Z
  GitCommit: 77556d9e64304c27c718bb0794676713628e435e
  GitTreeState: clean
  GoVersion: go1.20.6
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v5.1.0 2023-06-19T16:58:18Z
  Helm Version: v3.12.1+gf32a527
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.20.0

**Comment by crenshaw-dev** [comment]
Are you using standard Kustomize support, or Kustomize in a CMP?

**Comment by jstolfer** [comment]
We are using ArgoCD Vault Plugin
https://github.com/argoproj-labs/argocd-vault-plugin

**Comment by crenshaw-dev** [comment]
I bet you got hit by this: https://github.com/argoproj/argo-cd/issues/15339

**Comment by jstolfer** [comment]
I agree I do believe that is what happened; however, the kustomize file for the cluster in question wasn't changed during the upgrade process. Any idea why the repo-server and the plugin would have temporarily seen a bad/broken output from `kustomize build` ? It's almost like it was attempting to build from an old commit.

**Comment by crenshaw-dev** [comment]
It's a good question... I'm not sure about 2.8.3, but it's possible that the CMP sidecar logged an error.

**Comment by jgwest** [comment]
Based on comments, has this been resolved?

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and let us know if the issue is still present, please?
